code/game/machinery/droneDispenser.dm

```diff
@@ -125,6 +125,16 @@
 
 /obj/machinery/drone_dispenser/examine(mob/user)
 	. = ..()
+	var/material_requirement_string = "It needs "
+	if (iron_cost > 0)
+		material_requirement_string += "[iron_cost] iron "
+	if (iron_cost > 0 && glass_cost > 0)
+		material_requirement_string += "and "
```

Comment:
```suggestion
	if (iron_cost > 0)
		material_requirement_string += "[iron_cost] iron "
		if (glass_cost > 0)
			material_requirement_string += "and "
```
You don't need to check iron_cost again, just use tabs to bring it in scope of your first check